vllm/attention/backends/flashinfer.py

```diff
@@ -527,6 +631,20 @@ def prepare(self):
         self.total_blocks = 0
         self.is_profile_run: bool = False
 
+        if self.global_hyperparameters is None:
+            # Infer global hyperparameters, since currently we only support
+            # models in which all layers share the same values for the
+            # following hyperparameters:
+            # - `window_left`
+            # - `logits_soft_cap`
+            # - `sm_scale`
+            model = self.runner.model
```

Comment:
`vllm_config.compilation_config.static_forward_context` is a dict of layer prefix to attention layer. you can collect sliding window, etc. from there. no need to iterate over model's submodule.